src/main/java/au/org/ala/biocache/web/DownloadController.java

```diff
@@ -169,23 +176,65 @@ public boolean apply(Object source, String name, Object value) {
 
         DownloadDetailsDTO.DownloadType downloadType = hasDBColumn ? DownloadDetailsDTO.DownloadType.RECORDS_DB : DownloadDetailsDTO.DownloadType.RECORDS_INDEX;
 
-        return download(requestParams, ip, getUserAgent(request), apiKey, response, request, downloadType);
+        return download(requestParams, ip, getUserAgent(request), apiKey, email, response, request, downloadType);
     }
 
-    private Object download(DownloadRequestParams requestParams, String ip, String userAgent, String apiKey,
+    private Object download(DownloadRequestParams requestParams, String ip, String userAgent, String apiKey, String email,
                             HttpServletResponse response, HttpServletRequest request,
                             DownloadDetailsDTO.DownloadType downloadType) throws Exception {
 
+        // check the email is supplied and a matching user account exists with the required privileges
+        if (StringUtils.isEmpty(email)) {
+
+            response.sendError(HttpServletResponse.SC_PRECONDITION_FAILED, "Unable to perform an offline download without an email address");
+            return null;
+
+        }
+
+        // lookup the user details and check privileges based on the supplied email
+        try {
+
+            Map<String, ?> userDetails = authService.getUserDetails(email);
+
+            if (userDetails == null) {
+
+                response.sendError(HttpServletResponse.SC_FORBIDDEN, "Unable to perform an offline download matching user details");
+                return null;
+            }
+
+            boolean activated = (Boolean) userDetails.get("activated");
```

Comment:
It could be useful to switch to `userDetails.getOrDefault("activated", false)` here for clarity and to avoid having to rely on `null` being cast to `false` (which is not a typical Java expectation).